OutOfSchool/OutOfSchool.Common/Validators/CustomAgeAttribute.cs

```diff
@@ -96,4 +90,22 @@ private void EnsureThatMinAndMaxAgesAreValid()
             throw new InvalidOperationException("MaxAge cannot be less than MinAge.");
         }
     }
+
+    /// <summary>
+    /// Calculates min and max possible date.
+    /// </summary>
+    /// <param name="currentDate">Current date.</param>
+    /// <returns>Min and max possible dates.</returns>
+    private (DateOnly Min, DateOnly Max) CalculateMinAndMaxDate(DateOnly currentDate)
+    {
+        var maxYearOffset = currentDate.Year - 1;
+
+        var minAge = int.Clamp(MinAge, 0, maxYearOffset);
+        var maxAge = int.Clamp(MaxAge, 0, maxYearOffset);
+
+        var minDate = currentDate.AddYears(-maxAge);
```

Comment:
(Logic) I believe that `MaxAge` age should be allowed and we should disallow age starting from `MaxAge + 1`. Thus I expect here to have `.AddYears(-maxAge - 1)`. 

Ofc, this should be tested. And I've rechecked related tests and didn't find any check for age of person having tomorrow `maxAge+1` (it has to be allowed today, but denied tomorrow). Now tests are checking birthday dates precisely. That's why your current `IsValid_WhenDateIsExactlyMaxAge_ShouldReturnTrue` test is green but not covering the edge cases.

Reply:
> (Logic) I believe that `MaxAge` age should be allowed and we should disallow age starting from `MaxAge + 1`. Thus I expect here to have `.AddYears(-maxAge - 1)`.
> 
> Ofc, this should be tested. And I've rechecked related tests and didn't find any check for age of person having tomorrow `maxAge+1` (it has to be allowed today, but denied tomorrow). Now tests are checking birthday dates precisely. That's why your current `IsValid_WhenDateIsExactlyMaxAge_ShouldReturnTrue` test is green but not covering the edge cases.

Oh I got it, but I think that you mean something like `.AddYears(-maxAge).AddDays(-1)`, because if add one year to `MaxAge` then we allow every date that ends up to `MaxAge + 1` which is not correct, but I can be wrong.

As for day before `MinAge` and day after `MaxAge` I will fix this and create tests for these cases.

Reply:
Closing this one, though it's not resolved. See other comment below